### PR TITLE
[attestation] [QPL] Add Intel QPL's newer prototype for setting logging callback 

### PR DESCRIPTION
**Pull request by RRathna**
This PR adds Intel QPL's newer prototype for setting logging callback while keeping openenclave compatible with az-dcap-client

### CODE REVIEW

**Comment by RRathna** [comment]
bors try

**Comment by RRathna** [comment]
bors try

**Comment by RRathna** [comment]
bors try

File: host/sgx/sgxquoteprovider.c [hunk 1]
```diff
@@ -29,11 +29,8 @@ void oe_quote_provider_log(sgx_ql_log_level_t level, const char* message)
 {
     char formatted[OE_LOG_MESSAGE_LEN_MAX] = {0};
     oe_log_level_t oe_log_level[] = {
-        OE_LOG_LEVEL_ERROR,
-        OE_LOG_LEVEL_WARNING,
-        OE_LOG_LEVEL_INFO,
-        OE_LOG_LEVEL_NONE};
-    const char* dcap_level_strings[] = {"ERROR", "WARN", "INFO", "NONE"};
+        OE_LOG_LEVEL_ERROR, OE_LOG_LEVEL_INFO, OE_LOG_LEVEL_NONE};
+    const char* dcap_level_strings[] = {"ERROR", "INFO", "NONE"};
```
Comment (mingweishih) [inline host/sgx/sgxquoteprovider.c:33]: Does this mean there is no `WARNING` level in the QPL? What about the Azure-Dcap case?

**Review by mingweishih — COMMENTED** [review]
The changes look good to me. Just a question on removing `WARNING` level.

File: host/sgx/sgxquoteprovider.c (hunk 1 above)
Comment (RRathna) [reply host/sgx/sgxquoteprovider.c:33]: Yes, there are only `ERROR` and `INFO` levels in Intel QPL. Az-dcap should follow the public API definitions set by Intel QPL.

**Review by mingweishih — APPROVED** [review]

**Comment by RRathna** [comment]
bors try

**Comment by anakrish** [comment]
@RRathna The changes look good. However, the upgrade experience is not clear. 

1. What would happen if the user were to run the newer version of OE on a machine with older qpl?
2. Is there a way to inform the user in the above scenario?

**Comment by RRathna** [comment]
> @RRathna The changes look good. However, the upgrade experience is not clear.
> 
> 1. What would happen if the user were to run the newer version of OE on a machine with older qpl?


Hi @anakrish, Intel has made `sgx_ql_set_logging_callback_t` backward compatible by making the newly introduced parameter `sgx_ql_log_level_t ` optional, with a default value. (https://github.com/RRathna/openenclave/blob/qpl_logging_changes_2/host/sgx/platformquoteprovider.h#L92-L95)

So, both new and old versions of OE can work with both new and old versions of QPL and az-dcap(same as old qpl) after this change.
old OE + old QPL/az-dcap-client : `sgx_ql_set_logging_function_t`
old OE + new QPL : `sgx_ql_set_logging_function_t`
new OE + old QPL/az-dcap-client : `sgx_ql_set_logging_function_t`
new OE + new QPL : `sgx_ql_set_logging_callback_t`


> 2. Is there a way to inform the user in the above scenario?
In log level INFO, the prototype of logging function set is printed out in these line - https://github.com/RRathna/openenclave/blob/qpl_logging_changes_2/host/sgx/linux/sgxquoteproviderloader.c#L113-L127

**Review by anakrish — APPROVED** [review]
LGTM.
It would be good to ensure that the 4 scenarios you mention actually work as expected.

**Comment by radhikaj** [comment]
bors try

**Pull request merged**